### PR TITLE
Track incoming messages by org and label

### PR DESCRIPTION
**Pull request by rowanseymour**
Keeps track of number of incoming messages received each day using `DailyCount`. If labels on existing messages are manually changed after message is received, counts are updated accordingly.

### CODE REVIEW

File: casepro/msgs/tests.py [hunk 1]
```diff
@@ -308,8 +308,8 @@ def test_save(self):
             setattr(message, '__data__labels', [("L-001", "Spam")])
             message.save()
 
-        # check removing a group and adding new ones
-        with self.assertNumQueries(6):
+        # check removing a label and adding new ones
+        with self.assertNumQueries(10):
```
Comment (rowanseymour) [inline casepro/msgs/tests.py]: This now includes the `DailyCount` inserts

File: casepro/statistics/migrations/0003_auto_20160701_1412.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    dependencies = [
+        ('statistics', '0002_populate_reply_counts'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='dailycount',
+            name='count',
+            field=models.IntegerField(),
```
Comment (rowanseymour) [inline casepro/statistics/migrations/0003_auto_20160701_1412.py]: This was a `PositiveIntegerField` but now we need to record a message losing a label as a negative incremental count

File: casepro/statistics/migrations/0004_populate_incomining_counts.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import six
+
+from collections import defaultdict
+from django.db import migrations
+
+from dash.utils import chunks
+
+
+def populate_incoming_counts(apps, schema_editor):
```
Comment (rowanseymour) [inline casepro/statistics/migrations/0004_populate_incomining_counts.py]: Takes almost an hour on current UPartners database

**Comment by nicpottier** [comment]
yarp

**Pull request merged**